core/vm/evm.go

```diff
@@ -426,11 +426,17 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gasRemainin
 	ret, err = run(evm, contract, nil, false)
 
 	// EIP-170: Contract code size limit
-	if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > params.MaxCodeSize {
+	if err == nil && evm.chainRules.IsSpuriousDragon {
 		// Gnosis Chain prior to Shanghai didn't have EIP-170 enabled,
 		// but EIP-3860 (part of Shanghai) requires EIP-170.
 		if !evm.chainRules.IsAura || evm.config.HasEip3860(evm.chainRules) {
-			err = ErrMaxCodeSizeExceeded
+			if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
+				if len(ret) > params.MaxCodeSizePostAhmedabad {
```

Comment:
this introduces deep nesting that can be avoided if we extract this into a `func (evm *EVM) maxCodeSize()` function
e.g.

```
func (evm *EVM) maxCodeSize() int { 
    if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
        return params.MaxCodeSizePostAhmedabad
    }

    return params.MaxCodeSize
}
```

then the if remains unchanged:
```
if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > evm.maxCodeSize() {
    // same as before
}
```

Reply:
[Done](https://github.com/erigontech/erigon/pull/11792/commits/d29448ee5046fefa2886f6c63d0f9cad831e14f6), thank you!